contracts/Genesis.sol

```diff
@@ -196,31 +205,25 @@ contract Genesis is ERC721, VRFConsumerBase, Ownable {
     }
 
     /**
-     * Enumerable
-     */
-    // TODO Might want to add this one at some point. We would probably need a method to know the token types too.
-    // function tokensOfOwner(address _owner)
-    //     external
-    //     view
-    //     returns (uint256[] memory)
-    // {
-    //     uint256 tokenCount = balanceOf(_owner);
-    //     uint256[] memory tokensId = new uint256[](tokenCount);
-    //     for (uint256 i = 0; i < tokenCount; i++) {
-    //         tokensId[i] = tokenOfOwnerByIndex(_owner, i);
-    //     }
-
-    //     return tokensId;
-    // }
-
-    /**
-     * Withdrawing
+     * Withdraw balance from the contract
      */
-    // TODO Add the logic to distribute funds amongst the team
-    function withdraw() public payable onlyOwner {
+    function withdrawAll() public payable onlyOwner {
         uint256 balance = address(this).balance;
         require(balance > 0, "No ether left to withdraw");
         (bool success, ) = (msg.sender).call{value: balance}("");
         require(success, "Transfer failed.");
     }
+
+    /**
+     * Modifier that limits the function to WL members only
+     * @param nonce nonce to be used
+     * @param proof proof
+     * Owner bypasses this (for tests)
+     */
+    modifier onlyWhitelist(uint256 nonce, bytes32[] memory proof) {
+        if (msg.sender != owner()) {
+            require(verifyProof(nonce, proof), "Address is not in the whitelist");
```

Comment:
hmm, the message is a bit confusing here. I'm not 100% against, but wouldn't `Invalid proof` makes more sense? `Invalid proof, address is not whitelisted` or something if you want maybe?

Reply:
Also, here you only validate that the proof is right. Shouldn't you also validate that `proof === msg.sender`? Otherwise what tells me the `proof` wasn't spoofed?

Reply:
For the message, don't forget that it's only gonna happen if someone submits an invalid proof manually via the contract. Basically, if someone is trying to "cheat" the WL. So I really don't care about the message here. I can change it if you want.
about `msg.sender`, `verifyProof` verify the proof against a leaf that is created with `msg.sender`, so no spoof possible here.